### PR TITLE
Current reservations to map

### PR DESCRIPTION
**Pull request by willmt80**
Reserve new header:
reserve new

Reserve new modal:
reserve new modal

Current reservations with edit button:
current reservations v edit

Edit blocks header:
edit blocks header

### CODE REVIEW

File: src/router/index.js [hunk 1]
```diff
@@ -14,11 +14,11 @@ import AvailableTeams from '../views/AvailableTeams.vue';
 
 Vue.use(VueRouter);
 
-function getReserveRoute(route) {
-  return {
-    reservedFilter: route.params.editmode === 'edit' ? 1 : 0,
-  };
-}
+// function getReserveRoute(route) {
```
Comment (jackblanc) [inline src/router/index.js:17]: Any reason this shouldn't get deleted?

File: src/views/CurrentReservations.vue [hunk 2]
```diff
@@ -12,23 +12,29 @@
...
-    streets: [
-      { name: 'Abraham', type: 'ST' },
-      { name: 'Hemenway', type: '' },
-      { name: 'Hunington', type: 'BLVD' },
-      { name: 'Parker', type: 'AVE' },
-    ],
-  }),
+  computed: {
+    ...mapState(['reservedStreets']),
+    reservedStreets() {
```
Comment (jackblanc) [inline src/views/CurrentReservations.vue:29]: Do lines 28 and 29 not duplicate each other? I might be reading this wrong. @bensenescu

File: src/router/index.js [hunk 3]
```diff
@@ -68,9 +68,14 @@ const routes = [
   },
   {
     path: '/reserve/:editmode',
-    name: 'reserve',
+    name: 'ReserveNew',
     component: MapPage,
-    props: getReserveRoute,
+  },
+  {
+    path: '/reserve/:editmode',
```
Comment (jackblanc) [inline src/router/index.js:75]: Can you add a comment for allowed values for editmode somewhere in this file?

File: src/views/MapPage.vue [hunk 4]
```diff
@@ -66,20 +90,50 @@ export default {
...
+      const fids = [];
+      this.reservedStreets.forEach((street) => {
+        fids.push(street.FID);
+      });
+      return fids;
+    },
     async reserveStreets() {
       // reserve streets backend call
       // .then(
       this.$bvModal.show('street-confirmation-modal');
```
Comment (jackblanc) [inline src/views/MapPage.vue:127]: Is there anything that's preventing these three methods from being connected?

File: src/views/Map.vue [hunk 5]
```diff
@@ -141,13 +146,12 @@ export default {
         this.view.popup.on('trigger-action', (event) => {
           // Execute the measureThis() function if the measure-this action is clicked
           if (event.action.id === 'reserve-this') {
-            this.pushStreet(event.target.selectedFeature.attributes.FID);
+            this.pushStreet(event.target.selectedFeature.attributes.FID, 'reserve');
```
Comment (jackblanc) [inline src/views/Map.vue:149]: Can you write comments here?

File: src/views/Map.vue [hunk 6]
```diff
@@ -78,8 +82,11 @@ export default {
       ],
     };
     const sqlExpressions = [];
-    if (this.reservedFilter !== undefined) {
+    const reservedFidsFilter = `FID = ${this.fids.join(' OR FID = ')}`;
```
Comment (jackblanc) [inline src/views/Map.vue:85]: This logic is pretty complicated, can you add comments here?

**Comment by jackblanc** [comment]
Can you add some spacing between Currently Selected: 1234 and the Reserve button on the map page? Maybe center it under the title, on row with the reserve button. Not sure what designs are supposed to be but trying to find small things like these before we demo.

**Comment by willmt80** [comment]
> Looks awesome, I have a few questions though that I left.
> 
> Is it expected behavior that "complete" returns an error when you click it?

Did you get an error when clicking the button on the top of the page or when clicking complete on the popup? If a street is already in the unreserve row then you cannot move it to the completed row. We should probably include a way to change or remove the status of a selected street it just hasn't been designed or implemented yet.

File: src/views/MapPage.vue (hunk 4 above)
Comment (willmt80) [reply src/views/MapPage.vue:127]: Not as it stands. Once this has been connected to the backend then there will be three different ajax calls so I figured 3 separate methods might be better than one large method with a switch statement

Comment (gabriel-peter) [reply src/views/MapPage.vue:127]: Nice, I agree

Comment (jackblanc) [reply src/views/MapPage.vue:127]: I meant like implementing the requests to the backend itself

**Pull request merged**